apis/apiextensions/v1/composition_types.go

```diff
@@ -182,31 +199,179 @@ type ConnectionDetail struct {
+	// +optional
+	Resources *ContainerFunctionResources `json:"resources,omitempty"`
+
+	// Runner configuration for the Composition Function.
+	// +optional
+	Runner *ContainerFunctionRunner `json:"runner,omitempty"`
+
+	// TODO(negz): Should we support Env and EnvFrom? Presumably there's less
+	// reason to do so given it's possible to supply an arbitrary config object.
+}
+
+// A ContainerFunctionNetworkPolicy specifies the network policy under which
+// a containerized Composition Function will run.
+type ContainerFunctionNetworkPolicy string
+
+const (
+	// ContainerFunctionNetworkPolicyIsolated specifies that the Composition
+	// Function will not have network access; i.e. invoked inside an isolated
+	// network namespace.
+	ContainerFunctionNetworkPolicyIsolated ContainerFunctionNetworkPolicy = "Isolated"
+
+	// ContainerFunctionNetworkPolicyRunner specifies that the Composition
+	// Function will have the same network access as its runner, i.e. share its
+	// runner's network namespace.
+	ContainerFunctionNetworkPolicyRunner ContainerFunctionNetworkPolicy = "Runner"
+)
+
+// ContainerFunctionNetwork represents configuration for a Composition Function.
+type ContainerFunctionNetwork struct {
+	// Policy specifies the network policy under which the Composition Function
+	// will run. Defaults to 'Isolated' - i.e. no network access. Specify
+	// 'Runner' to allow the function the same network access as
+	// its runner.
+	// +optional
+	// +kubebuilder:validation:Enum="Isolated";"Runner"
+	// +kubebuilder:default=Isolated
+	Policy *ContainerFunctionNetworkPolicy `json:"policy,omitempty"`
+}
+
+// ContainerFunctionResources represents compute resources that may be used by a
+// Composition Function.
+type ContainerFunctionResources struct {
+	// Limits specify the maximum compute resources that may be used by the
+	// Composition Function.
+	// +optional
+	Limits *ContainerFunctionResourceLimits `json:"limits,omitempty"`
+
+	// NOTE(negz): We don't presently have any runners that support scheduling,
+	// so we omit Requests for the time being.
+}
+
+// ContainerFunctionResourceLimits specify the maximum compute resources
+// that may be used by a Composition Function.
+type ContainerFunctionResourceLimits struct {
+	// CPU, in cores. (500m = .5 cores)
+	// +kubebuilder:default="100m"
+	// +optional
+	CPU *resource.Quantity `json:"cpu,omitempty"`
+
+	// Memory, in bytes. (500Gi = 500GiB = 500 * 1024 * 1024 * 1024)
+	// +kubebuilder:default="128Mi"
+	// +optional
+	Memory *resource.Quantity `json:"memory,omitempty"`
+}
+
+// ContainerFunctionRunner represents runner configuration for a Composition
+// Function.
+type ContainerFunctionRunner struct {
+	// TODO(negz): url.Parse can't handle the @ in the below address.
+
+	// Endpoint specifies how and where Crossplane should reach the runner it
+	// uses to invoke containerized Composition Functions.
+	// +optional
+	// +kubebuilder:default="unix-abstract://@crossplane/fn/default.sock"
```

Comment:
I need to remove this TODO and update this to match `DefaultTarget`.

Reply:
Also, interesting to note that this kind of means we have "webhook" support already. Crossplane will send a gRPC request with a bunch of stuff instructing the function runner how to run a container function, but I suppose there's nothing stopping the server ignoring all of that and just doing something with the FunctionIO.